coveralls/api.py

```diff
@@ -76,67 +72,79 @@ def ensure_token(self):
     @staticmethod
     def load_config_from_appveyor():
         pr = os.environ.get('APPVEYOR_PULL_REQUEST_NUMBER')
-        return 'appveyor', os.environ.get('APPVEYOR_BUILD_ID'), pr
+        return 'appveyor', os.environ.get('APPVEYOR_BUILD_ID'), None, pr
 
     @staticmethod
     def load_config_from_buildkite():
         pr = os.environ.get('BUILDKITE_PULL_REQUEST')
         if pr == 'false':
             pr = None
-        return 'buildkite', os.environ.get('BUILDKITE_JOB_ID'), pr
+        return 'buildkite', os.environ.get('BUILDKITE_JOB_ID'), None, pr
 
     @staticmethod
     def load_config_from_circle():
         pr = os.environ.get('CI_PULL_REQUEST', '').split('/')[-1] or None
-        return 'circle-ci', os.environ.get('CIRCLE_BUILD_NUM'), pr
+        return 'circle-ci', os.environ.get('CIRCLE_BUILD_NUM'), None, pr
 
-    @staticmethod
-    def load_config_from_github():
-        service_number = os.environ.get('GITHUB_SHA')
+    def load_config_from_github(self):
+        service = 'github'
+        if self.config.get('repo_token'):
+            service = 'github-actions'
+        else:
+            gh_token = os.environ.get('GITHUB_TOKEN')
+            if not gh_token:
+                raise CoverallsException(
+                    'Running on Github Actions but GITHUB_TOKEN is not set. '
+                    'Add "env: GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}" to '
+                    'your step config.')
+            self.config['repo_token'] = gh_token
+
+        job = os.environ.get('GITHUB_RUN_NUMBER')
```

Comment:
I suggest to use `None` here instead. `GITHUB_RUN_NUMBER` is not unique at all, but `job` is being passed to `service_job_id` which should be [`A unique identifier of the job on the service specified by service_name.`](https://docs.coveralls.io/api-reference) But `GITHUB_RUN_NUMBER` isn't unique: [`A unique number for each run of a particular workflow in a repository. This number begins at 1 for the workflow's first run, and increments with each new run. This number does not change if you re-run the workflow run.`](https://docs.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables), it's just a number of the workflow run, it doesn't change for the individual parallel job runs. And there doesn't seem to be any other variable that does identify the individual job run, so our only option is not to pass `service_job_id` at all and let coveralls generate it itself. I tested that and it works that way.